src/npz.rs

```diff
@@ -243,3 +253,545 @@ impl<R: Read + Seek> NpzReader<R> {
+///   to ensure they have a valid bit pattern.
+///
+/// - The `.npy` files within the `.npz` archive must be at least 16-byte or
+///   ideally 64-byte aligned. Archives not created by this crate can be aligned
+///   with the help of the CLI tool [`rezip`] as in `rezip in.npz -o out.npz`.
+///
+/// [`rezip`]: https://crates.io/crates/rezip
+///
+/// # Example
+///
+/// This is an example of opening an immutably memory-mapped `.npz` archive as
+/// an [`NpzView`] providing an [`NpyView`] for each uncompressed `.npy` file
+/// within the archive which can be accessed via [`NpyView::view`] as
+/// immutable [`ArrayView`].
+///
+/// This example uses the [`memmap2`](https://crates.io/crates/memmap2) crate
+/// because that appears to be the best-maintained memory-mapping crate at the
+/// moment, but [`Self::new`] takes a `&mut [u8]` instead of a file so that you
+/// can use the memory-mapping crate you're most comfortable with.
+///
+/// ```
+/// use std::fs::OpenOptions;
+/// use memmap2::MmapOptions;
+/// use ndarray_npy::{NpzView, ViewNpzError};
+/// use ndarray::Ix1;
+///
+/// // Open `.npz` archive of uncompressed `.npy` files in native endian.
+/// #[cfg(target_endian = "little")]
+/// let file = OpenOptions::new().read(true)
+///     .open("tests/examples_little_endian_64_byte_aligned.npz").unwrap();
+/// #[cfg(target_endian = "big")]
+/// let file = OpenOptions::new().read(true)
+///     .open("tests/examples_big_endian_64_byte_aligned.npz").unwrap();
+/// // Memory-map `.npz` archive of 64-byte aligned `.npy` files.
+/// let mmap = unsafe { MmapOptions::new().map(&file).unwrap() };
+/// let npz = NpzView::new(&mmap)?;
+/// // List uncompressed files only.
+/// for npy in npz.names() {
+///     println!("{}", npy);
+/// }
+/// // Get immutable `.npy` views.
+/// let x_npy_view = npz.by_name("i64.npy")?;
+/// let y_npy_view = npz.by_name("f64.npy")?;
+/// // Optionally verify CRC-32 checksums.
+/// x_npy_view.verify()?;
+/// y_npy_view.verify()?;
+/// // Get and print immutable `ArrayView`s.
+/// let x_array_view = x_npy_view.view::<i64, Ix1>()?;
+/// let y_array_view = y_npy_view.view::<f64, Ix1>()?;
+/// println!("{}", x_array_view);
+/// println!("{}", y_array_view);
+/// # Ok::<(), ViewNpzError>(())
+/// ```
+#[derive(Debug, Clone)]
+pub struct NpzView<'a> {
+    files: HashMap<usize, NpyView<'a>>,
+    names: HashMap<String, usize>,
+}
+
+impl<'a> NpzView<'a> {
+    /// Creates a new immutable view of a memory-mapped `.npz` file.
+    pub fn new(bytes: &'a [u8]) -> Result<Self, ViewNpzError> {
+        let mut zip = ZipArchive::new(Cursor::new(bytes))?;
+        let mut files = HashMap::new();
+        let mut names = HashMap::new();
+        let mut index = 0;
+        for zip_index in 0..zip.len() {
+            let file = zip.by_index(zip_index)?;
+            // Skip directories and compressed files.
+            if file.is_dir() || file.compression() != CompressionMethod::Stored {
```

Comment:
Sounds good. As `zip` supports encrypted files wich cannot be memory-mapped, I've updated the code and docs already to skip encrypted files as well. It would make sense to complemented `compressed_names()` with `encrypted_names()`.

Reply:
And also `directory_names()`. Done.